### PR TITLE
Multi-lingual support

### PR DESCRIPTION
**Pull request by andrenatal**
@milescrabill @g-k r?

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -210,7 +210,8 @@ app.post('*', function (req, res, next) {
       Body: req.body,
       Bucket: config.s3_bucket,
       ContentType: 'audio/opus',
-      Key: key_base + '/audio.opus'
+      Key: key_base + '/audio.opus',
+      Headers: req.headers
```
Comment (milescrabill) [inline server.js]: What's the intent behind passing these headers through to S3? This also doesn't appear to be a supported parameter: https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/S3.html#putObject-property.

Comment (andrenatal) [reply server.js]: we need to save the language to the meta file, but seems this is not the right way

Comment (andrenatal) [reply server.js]: seems I'll need to add something to resbody then: https://github.com/mozilla/speech-proxy/blob/master/server.js#L287

Comment (g-k) [reply server.js]: Yeah, I'd say pull the headers we want out and validate them against a whitelist of known, expected values.

File: server.js [hunk 2]
```diff
@@ -253,7 +254,7 @@ app.post('*', function (req, res, next) {
     url: config.asr_url,
     method: 'POST',
     body: opusdec.stdout,
-    headers: {'Content-Type': 'application/octet-stream'},
+    headers: {'Content-Type': 'application/octet-stream', 'Language': req.headers.language},
```
Comment (g-k) [inline server.js]: Should validate this too and test that 1) invalid value languages are rejected and 2) add a valid language to the happy/success path

Comment (andrenatal) [reply server.js]: I think this validation should be done in the speech server itself, instead here, which is only the middleware. I'll prepare a PR for the speech server and share with you

File: server.js (hunk 1 above)
Comment (andrenatal) [reply server.js]: I'm working on a PR to save the metadata in a different file

**Comment by andrenatal** [comment]
@milescrabill @g-k r?

So here is a new version of the patch containing:

- Multi-lingual support: The client now can choose which language he wants to have his speech decoded by passing the parameter `Accept-Language` in the request header.  I agree we need to have a whitelist, but I'd prefer to have that in the speech server instead having it here, which is supposed to only be a proxy and not interfere in the request.  LMK what you guys think.  

- Introduces the support for two new headers fields which permit the client to allow or not the store of the audio sample and transcription `(Store-Transcription and Store-Sample)`

- Now storing the request metadata containing the `User-Agent` and a tag containing the product name `(Product-Tag)` that were passed by the client in the request header. We need those to identify the products using the speech API.

File: server.js (hunk 1 above)
Comment (andrenatal) [reply server.js]: fixed

File: server.js (hunk 2 above)
Comment (andrenatal) [reply server.js]: We are already defaulting to english here: https://github.com/mozilla/gspeech/commit/6367878e1950909c017b12cdd8b85a51574edb04  but I think a white list is important even though I have some concerns, but I prefer to have that and discuss in the speech-to-text server.

**Comment by andrenatal** [comment]
Travis is failing because the linter is panicking about the regex. 
```
/home/travis/build/mozilla/speech-proxy/server.js
  20:24  error  Unexpected control character(s) in regular expression: \x09  no-control-regex
  20:43  error  Unnecessary escape character: \/                             no-useless-escape
  20:45  error  Unnecessary escape character: \.                             no-useless-escape
```

File: languages.json [hunk 3]
```diff
@@ -0,0 +1,121 @@
+{
+  "af-ZA": "Afrikaans (Suid-Afrika)",
```
Comment (g-k) [inline languages.json]: :+1: nice

File: Makefile [hunk 4]
```diff
@@ -10,8 +10,8 @@ run:
 		speech-proxy:build
 
 test:
-	curl -i -X POST http://localhost:9001/ -H 'Content-Type: audio/opus' --data-binary @speech_orig.opus
-	curl -i -X POST http://localhost:9001/ -H 'Content-Type: audio/webm' --data-binary @webm.webm
+	curl -i -X POST http://localhost:9001/ -H 'Content-Type: audio/opus' -H 'Accept-Language: en-us' -H 'Store-Sample: 1' -H 'Store-Transcription: 1' -H 'Product-Tag: shell-curl' --data-binary @speech_orig.opus
+	curl -i -X POST http://localhost:9001/ -H 'Content-Type: audio/webm' -H 'Accept-Language: en-us' -H 'Store-Sample: 1' -H 'Store-Transcription: 1' -H 'Product-Tag: shell-curl' --data-binary @webm.webm
```
Comment (g-k) [inline Makefile]: nit: it'd be good to test with different metadata options too:

* success with store-sample 0
* success without store-sample
* success with store-transcription 0
* success without store-transcription
* failure for invalid accept-language
* failure for invalid product-tag
* failure for invalid store-sample
* failure for invalid store-transcription

I think we can test failure (at least that it returns the right status code) as follows:

```
curl -s -o /dev/null -w "%{http_code}" -i -X POST http://localhost:9001/ -H 'Content-Type: audio/webm' -H 'Accept-Language: \0\0' | grep 400 
```

**Pull request merged**